### DISCUSSION TITLE
Vite + Svelte / Sveltekit

### DISCUSSION THREAD

**Discussion by userquin** [unocss/unocss#175] [General]
For issue #6: we should work here, I don't move the issue to keep track the integration

It seems the svelte vite plugin should be before unocss plugin, on first run there is no problem inverting the order, but for HMR the order matters: if I configure unocss  + svelt HMR stops working but then first request stops working, if I invert it, then first page works but HMR stops working.

I pushed to my main branch to revert the plugin order, also added some bg color test with svelte dynamic class:*** and some testing with `data-***` (not working).

@axmad386 @aewing @lkj4

**Comment by userquin** [comment]
you can check also `per-module` (scoped css) here https://github.com/userquin/unocss-svelte-ts/tree/per-module

In that branch I change the first logo to use the `logo shortcut` and seems to be working: I think there is a problem with svelte.

**Comment by userquin** [comment]
Talking with some expert on svelte, it seems we SHOULD add `svelte vite plugin` after `unocss vite plugin`, I update both branches.

**Comment by userquin** [comment]
can anyone take a look at the per-module branch I've update rn? It seems `unocss` is not generating the `[data-***]` style, only for class `.data-***`.

**Reply by aewing** [reply]
I can take a look in a couple of hours. I was definitely seeing some behavior around HMR that was different from what I had come to expect when working with Tailwind. I also find myself missing `@apply` a lot. I need to spend more time tinkering with UnoCSS, but I am contemplating some sort of utility component/action/preprocessor to augment unocss in svelte.

**Reply by userquin** [reply]
@aewing I'm playing on `unocss` repo with `per-module` and it seems promising, see https://github.com/antfu/unocss/issues/169#issuecomment-985941541

**Reply by userquin** [reply]
I added a working example with `absolute` and `class:foo={foo}` for `svelte` on https://github.com/antfu/unocss/tree/feat/per-module-import branch: `pnpm install && pnpm run build` from root, then go to `test/fixtures/vite-svelte` and run `pnpm install && pnpm run dev`.

**Reply by userquin** [reply]
tested also `pnpm run build && pnpm run serve` on `test/fixtures/vite-svelte`

**Reply by userquin** [reply]
added support for `class:foo` shorthand: https://imgur.com/fgzRc7L

**Reply by userquin** [reply]
added `sveltekit` playground
